src/Workspaces/Core/MSBuild/MSBuild/MSBuildWorkspace.cs

```diff
@@ -263,6 +335,64 @@ private static string GetAbsolutePath(string path, string baseDirectoryPath)
             return projectResult;
         }
 
+        /// <summary>
+        /// Reload the given projects based on their contents on disk.
+        /// </summary>
+        /// <param name="projectIds">The <see cref="ProjectId" /> set to be reloaded</param>
+        /// <param name="targets">The set of targets to execute on all projects in the solution</param>
+        /// <param name="msbuildLogger">An optional <see cref="ILogger"/> that will log msbuild results..</param>
+        /// <param name="progress">An optional <see cref="IProgress{T}"/> that will receive updates as the project is opened.</param>
+        /// <param name="cancellationToken">An optional <see cref="CancellationToken"/> to allow cancellation of this operation.</param>
+        public async Task<ImmutableArray<Project>> ReloadProjectsAsync(
+            ImmutableArray<ProjectId> projectIds,
+            ImmutableArray<string> targets = default,
+            ILogger? msbuildLogger = null,
+            IProgress<ProjectLoadProgress>? progress = null,
+            CancellationToken cancellationToken = default)
+        {
+            if (targets == default)
+            {
+                targets = Targets.Default;
+            }
+
+            using var _1 = ArrayBuilder<Project>.GetInstance(out var projectsToReload);
+            foreach (var projectId in projectIds)
+            {
+                this.CheckProjectIsInCurrentSolution(projectId);
```

Comment:
Does this need to have some sort of locking to ensure simultaneous calls don't do bad things? I'm not entirely sure what guarantees MSBuildWorkspace has made in the past...